kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -365,7 +368,10 @@ private TransactionRebaseState resolveConflicts(
         maxRetries);
     TransactionRebaseState rebaseState =
         ConflictChecker.resolveConflicts(engine, readSnapshot, commitAsVersion, this, dataActions);
-    long newCommitAsVersion = rebaseState.getLatestVersion() + 1;
+    long rebasedVersion = rebaseState.getLatestVersion();
+    this.currentCrcInfo =
```

Comment:
In the future should we also try to use this to have fileSizeHistogram? as future work is fine

Reply:
Good catch, added a TODO, I will send a PR to update `resetActionMetricsForRetry` to plumb current file size histogram and add tests.